extensions/wikia/FilePage/FilePageHooks.class.php

```diff
@@ -260,62 +257,32 @@ public static function onGetFileLinks( $id, &$links ) {
 	 * @param Title $title -- instance of Title class
 	 *
 	 */
-	private static function purgeTitle( Title $title ) {
-		if ( $title->inNamespace( NS_FILE ) ) {
-			self::purgeRedir( $title );
-		} else {
-			self::clearLinkedFilesCache( $title->getArticleID() );
-		}
-	}
-
-
-	/**
-	 * Clear memcache and purge page
-	 *
-	 * @param Title $title -- instance of Title class
-	 *
-	 */
-	private static function purgeRedir( Title $title ) {
+	private static function purgeMemcache( Title $title ) {
 		global $wgMemc;
 		$redirKey = wfMemcKey( 'redir', 'http', $title->getPrefixedText() );
 		$wgMemc->delete( $redirKey );
 		$redirKey = wfMemcKey( 'redir', 'https', $title->getPrefixedText() );
 		$wgMemc->delete( $redirKey );
-		$page = WikiPage::factory( $title );
-		$page->doPurge();
-	}
-
-
-	/**
-	 * getFileLinks get links to material
-	 *
-	 * @param $id Int: page_id value of the page being deleted
-	 *
-	 * @return ResultWrapper -  image links
-	 */
-	private static function getFileLinks( $id ) {
-		$dbr = wfGetDB( DB_SLAVE );
-
-		return $dbr->select(
-			[ 'imagelinks' ],
-			[ 'il_to' ],
-			[ 'il_from' => $id ],
-			__METHOD__,
-			[ 'ORDER BY' => 'il_to', ] );
 	}
 
 	/**
 	 * Clear memcache redirs before db changed
 	 *
 	 * @param $id Int: page_id value of the page being deleted
 	 */
-	private static function clearLinkedFilesCache( $id, $results = null ) {
-		if ( is_null( $results ) ) {
-			$results = self::getFileLinks( $id );
+	private static function clearLinkedFilesCache( Title $title  , bool $memcacheOnly=false) {
+		$results = FilePageHelper::getFileLinks( $title->getArticleID() );
+		$keys = FilePageHelper::getSurrogateKeys( $title );
+		if ( $results ) {
+			foreach ( $results as $row ) {
+				$title = Title::makeTitleSafe( NS_FILE, $row->il_to );
+				self::purgeMemcache( $title );
+			}
 		}
-		foreach ( $results as $row ) {
-			$title = Title::makeTitleSafe( NS_FILE, $row->il_to );
-			self::purgeRedir( $title );
+		if( !$memcacheOnly ){
+			foreach( $keys  as $key) {
+				Wikia::purgeSurrogateKey( $key );
```

Comment:
Can you please log the keys we're purging to kibana so we get a more info about the scope of purges when testing/releasing.